### PR TITLE
[SPARK-20506][DOCS] 2.2 migration guide

### PR DESCRIPTION
**Pull request by MLnick**
Update ML guide for migration `2.1` -> `2.2` and the previous version migration guide section.

## How was this patch tested?

Build doc locally.

### CODE REVIEW

**Comment by MLnick** [comment]
Just doing a final check to find any other behavior changes or stuff that was deprecated in this release cycle.

**Comment by MLnick** [comment]
cc @yanboliang @srowen @jkbradley

File: docs/ml-guide.md [hunk 1]
```diff
@@ -72,35 +72,26 @@ MLlib is under active development.
...
 
 ### Deprecations and changes of behavior
 
 **Deprecations**
 
-* [SPARK-18592](https://issues.apache.org/jira/browse/SPARK-18592):
-  Deprecate all Param setter methods except for input/output column Params for `DecisionTreeClassificationModel`, `GBTClassificationModel`, `RandomForestClassificationModel`, `DecisionTreeRegressionModel`, `GBTRegressionModel` and `RandomForestRegressionModel`
+There are no deprecations.
 
 **Changes of behavior**
```
Comment (yanboliang) [inline docs/ml-guide.md]: Should we include #17233 in this section?

**Review by yanboliang — COMMENTED** [review]
One minor question, otherwise, LGTM. Thanks!

File: docs/ml-guide.md (hunk 1 above)
Comment (MLnick) [inline docs/ml-guide.md]: Thanks - didn't catch that one

**Comment by MLnick** [comment]
@felixcheung  @yanboliang by the way I haven't added any SparkR stuff here as I'm not sure anything breaking, deprecated etc goes here or in the SparkR guide.

**Comment by yanboliang** [comment]
@MLnick We have separate SparkR guide at http://spark.apache.org/docs/latest/sparkr.html . Thanks.

**Comment by felixcheung** [comment]
looks good
We do have a separate section but it's mostly for breaking/behavior changes at the R layer, if there are changes to ml that affects R also would be great to include here.

and I think "what's new" is a great idea too

File: docs/ml-guide.md [hunk 2]
```diff
@@ -66,41 +66,59 @@ To use MLlib in Python, you will need [NumPy](http://www.numpy.org) version 1.4
...
+The list below highlights some of the new features and enhancements added to MLlib in the `2.2`
+release of Spark:
+
+* `ALS` methods for _top-k_ recommendations for all users or items, matching the functionality
+ in `mllib` ([SPARK-19535](https://issues.apache.org/jira/browse/SPARK-19535)). Performance
+ was also improved for both `ml` and `mllib`
+ ([SPARK-11968](https://issues.apache.org/jira/browse/SPARK-11968) and
+ [SPARK-20587](https://issues.apache.org/jira/browse/SPARK-20587))
+* `Correlation` and `ChiSquareTest` stats functions for `DataFrames`
+ ([SPARK-19635](https://issues.apache.org/jira/browse/SPARK-19635) and
```
Comment (dongjoon-hyun) [inline docs/ml-guide.md]: Hi, @MLnick . `Correlation` issue is SPARK-19636.

**Comment by dongjoon-hyun** [comment]
+1, LGTM except one minor comment about JIRA issue number.

File: docs/ml-guide.md (hunk 2 above)
Comment (MLnick) [inline docs/ml-guide.md]: Ah right thanks for catching that

**Comment by MLnick** [comment]
Ok, I think this is good to go so will merge shortly.

@felixcheung for R, if you do know of anything else that should go here (rather than in the SparkR migration guide) let me know.

**Comment by MLnick** [comment]
Merged to master/branch-2.2

**Comment by felixcheung** [comment]
right, I reviewed them 
- [this](https://github.com/apache/spark/pull/17996/files#diff-a9770b923a4959616bc2126d4afd61eaR35) in ML could also affect R
- [this](https://github.com/apache/spark/blame/master/docs/sparkr.md#L657) in R might be good to include in ML guide too

**Comment by MLnick** [comment]
@felixcheung ok - the KMeans behavior change was in 2.1 actually (so not in this migration guide, in this PR it was moved to the "old versions migration guides" page).

For `spark.lda` issue - was that only for SparkR API? Or for all of LDA? If the latter then it could be included perhaps. If the former then I think just in the R section of the guide, no need for it here.

**Comment by MLnick** [comment]
I actually realized I forgot to add the HTML links to the relevant guide sections in the highlight list - opened a little PR at #18043 to fix that.

**Pull request closed without merging**